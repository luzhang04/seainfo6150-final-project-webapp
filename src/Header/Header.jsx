import React from "react";

function Header() {
  return(
    <div>
    <h1>D.I.Y</h1>
    </div>
  )
}

export default Header;
