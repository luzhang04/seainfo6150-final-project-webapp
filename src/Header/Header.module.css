
.nav {
    display: flex;
    align-content: center;
    align-items: center;
    background-color: #81ecec;
    top: 0;
    width: 100%;
    overflow: hidden;
  
  }
  
  .logo {
    width: 100px;
    height: 100px;
    text-align: left;
    margin-left: 2em;
    margin-right: 15em;
    margin-top: 0.7em;
    margin-bottom:0.5em;
  
  } 
  
  .list {
    display: flex;
    list-style-type: none;
    text-align: center;
    align-content: center;
    align-items: center;
    justify-content: center;
    font-family: 'Montserrat', Helvetica, sans-serif;
    font-size: 1.7em;
  
  }
  
  .li {
    margin: auto 2em;
    display:inline;
  }
  
  .navlink {
    color: white;
    text-decoration: none;
  }

  a:hover {
    color :#fab1a0;
  }
  
  @media (max-width: 1600px) {
    .li {
      margin: auto 0.5em;
      display:inline;
    }
    .logo {
      margin-right: 5em;
    }
    .list {
      font-size: 1.5em;
      font-weight: bold;
    }
  }

  @media (max-width: 800px) {
    .li {
      margin: auto 0.5em;
      display:inline;
    }
    .logo {
      margin-right: 2em;
    }
    .list {
      font-size: 1em;
      font-weight: bold;
    }
  }