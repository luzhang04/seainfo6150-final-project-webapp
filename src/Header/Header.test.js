import React from "react";
import { render } from "@testing-library/react";
import Header from "./Header.jsx";

describe("Header tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Header />);
      expect(container).toMatchSnapshot();
    });
  });