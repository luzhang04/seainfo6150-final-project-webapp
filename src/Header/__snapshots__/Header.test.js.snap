// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header tests renders correctly 1`] = `
<div>
  <header
    class="nav"
  >
    <a
      href="/"
    >
      <img
        alt="diy logo"
        class="logo"
        src="https://media1.giphy.com/media/ZCSbHLCd1cdEYw7rw4/giphy.gif"
      />
    </a>
    <nav>
      <ul
        class="list"
      >
        <li
          class="li"
        >
          <a
            class="navlink"
            href="/diylist"
          >
            All crafts
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="navlink"
            href="/category"
          >
            Category
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="navlink"
            href="/subscribe"
          >
            Subscribe
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="navlink"
            href="/aboutus"
          >
            About us
          </a>
        </li>
        <li
          class="li"
        >
          <a
            class="navlink"
            href="/contactus"
          >
            Contact Us
          </a>
        </li>
      </ul>
    </nav>
  </header>
</div>
`;
